magma/ref.py

```diff
@@ -213,8 +199,33 @@ def qualifiedname(self, sep="."):
     def anon(self):
         return self.view.port.anon()
 
-    def parent(self):
-        return None
-
     def __str__(self):
         return str(self.view.port.name)
+
+    def root(self):
+        return self.view.root()
+
+
+def get_ref_inst(ref):
+    """
+    If value is part of a port on an instance, return that instance,
+    otherwise None
+    """
+    if not isinstance(ref.root(), InstRef):
+        return None
+    return ref.root().inst
```

Comment:
can cache `ref.root()` so we don't compute it twice

```suggestion
    """
    If value is part of a port on an instance, return that instance,
    otherwise None.
    """
    root = ref.root()
    if not isinstance(root, InstRef):
        return None
    return root.inst
```